src/js/Controllers/TTSController.js

```diff
@@ -167,40 +119,45 @@ class TTSController {
         }
     }
 
-    playNext(rpc) {
-        var file = this.filePlaylist[0];
+    playNext() {
+        var file = this.filePlaylist.shift();
         if (file !== undefined) {
             if (file.type === "FILE") {
-                this.playAudio(rpc);
+                this.playAudio(file.text);
             } else if (file.type === "TEXT") {
-                this.speak(rpc);
+                this.speak(file.text);
             } else if (file.type === "REPLY") {
                 // REPLY is not an HMI_API type, it is internal used
                 // in case more things will be added to the filePlaylist,
                 // so that TTS Speak will be replied to before speaking next message
-                this.filePlaylist.shift();
                 clearInterval(this.timers[file.id]);
                 this.speakEnded();
                 this.listener.send(RpcFactory.TTSSpeakSuccess(file.id));
+                return;
             }
+        } else {
+            this.speakEnded();
         }
     }
 
     isAlertSpeakInProgress() {
-        return this.speakType.includes('ALERT');
+        for(const file of this.filePlaylist) {
+            if(file.type === 'REPLY' && file.speakType.includes('ALERT')) return true;
+        }
+        return false;
     }
-
+    
```

Comment:
redundant spacing

Reply:
@YuriiYefimenko Did you read a  [comment](https://github.com/smartdevicelink/generic_hmi/pull/411#pullrequestreview-724360137) well ?
If so, please take a look changes on this [file](https://raw.githubusercontent.com/smartdevicelink/generic_hmi/develop/src/js/Controllers/TTSController.js) on develop branch.